File: src/main/java/org/jhapy/notification/domain/CloudNotificationMessage.java
package org.jhapy.notification.domain;

import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.ToString;
import org.springframework.data.mongodb.core.mapping.Document;

/**
 * @author jHapy Lead Dev.
 * @version 1.0
 * @since 2019-07-09
 */
@Document
@Data
@EqualsAndHashCode(callSuper = true)
@ToString(callSuper = true)
public class CloudNotificationMessage extends BaseEntity {

  private String deviceToken;

  private String title;

  private String body;

  private String data;

  private String cloudNotificationMessageAction;

  private CloudNotificationMessageStatusEnum cloudNotificationMessageStatus;

  private String errorMessage;

  private int nbRetry = 0;

  private String applicationName;

  private String iso3Language;
}

File: src/main/java/org/jhapy/notification/domain/CloudNotificationMessage.kt
package org.jhapy.notification.domain

import org.springframework.data.mongodb.core.mapping.Document

/**
 * @author jHapy Lead Dev.
 * @version 1.0
 * @since 2019-07-09
 */
@Document
class CloudNotificationMessage : BaseEntity() {
    var deviceToken: String? = null
    var title: String? = null
    var body: String? = null
    var data: String? = null
    var cloudNotificationMessageAction: String? = null
    var cloudNotificationMessageStatus: CloudNotificationMessageStatusEnum? = null
    var errorMessage: String? = null
    var nbRetry = 0
    var applicationName: String? = null
    var iso3Language: String? = null
}

enum class CloudNotificationMessageStatusEnum {
    NOT_SENT, SENT, ERROR, RETRYING
}
